services/src/contexts/schema.sql

```diff
@@ -634,11 +634,123 @@ CREATE TABLE collection_children (
     PRIMARY KEY (parent, child)
 );
 
+CREATE TYPE "ArunaDataProviderDefinition" AS (
+    id uuid,
+    "name" text,
+    api_url text,
+    project_id text,
+    api_token text,
+    filter_label text,
+    cache_ttl int
+);
+
+CREATE TYPE "DatabaseConnectionConfig" AS (
+    host text,
+    port int,
+    "database" text,
+    schema text,
+    "user" text,
+    "password" text
+);
+
+CREATE TYPE "GbifDataProviderDefinition" AS (
+    "name" text,
+    db_config "DatabaseConnectionConfig",
+    cache_ttl int
+);
+
+CREATE TYPE "GfbioAbcdDataProviderDefinition" AS (
+    "name" text,
+    db_config "DatabaseConnectionConfig",
+    cache_ttl int
+);
+
+CREATE TYPE "GfbioCollectionsDataProviderDefinition" AS (
+    "name" text,
+    collection_api_url text,
+    collection_api_auth_token text,
+    abcd_db_config "DatabaseConnectionConfig",
+    pangaea_url text,
+    cache_ttl int
+);
+
+CREATE TYPE "TextMetaDataDefinitionKeyValue" AS (
+    key text,
+    value "MetaDataDefinition"
+);
+
+CREATE DOMAIN "LayerId" AS text;
+
+CREATE DOMAIN "LayerCollectionId" AS text;
+
+CREATE TYPE "MockLayer" AS (
+    id "LayerId",
+    "name" text,
+    "description" text,
+    symbology "Symbology",
+    provenance "Provenance",
+    workflow json, -- TODO: get rid of json type
```

Comment:
is it difficult to get rid of id? Maybe a `mock_workflows` table that's the same as the  `workflows` table?

Reply:
[afacdd0](https://github.com/geo-engine/geoengine/pull/866/commits/afacdd00f469216470891236b82bbec8bb532ea0)